task1.cpp

```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+#include <string>
+#include <vector>
+
+//KMP using prefix function
+void KMP(std::string str, std::string pattern) {
+    int size_str = str.size();
+    int size_pattern = pattern.size();
+    std::vector<int> prefix(size_pattern);
+    prefix[0] = 0;
+    int j = 0;
+    for (int i = 1; i < prefix.size(); ++i) {
+        if (pattern[i] == pattern[j]) {
+            j++;
+            prefix[i] = j;
+            continue;
+        } else {
+            if (j == 0) {
+                prefix[i] = 0;
+                continue;
+            } else {
+                j = prefix[j - 1];
+            }
+        }
+    }

```

Comment:
counting the prefix vector should be done in a separate function

Reply:
I tried but if i calculated in seperate function, i got TL

Reply:
I have fixed it